sdk/azcore/arm/runtime/pipeline.go

```diff
@@ -43,7 +46,20 @@ func NewPipeline(module, version string, cred azcore.TokenCredential, plOpts azr
 		copy(perCall, plOpts.PerCall)
 		plOpts.PerCall = append(perCall, regPolicy)
 	}
-	return azruntime.NewPipeline(module, version, plOpts, &options.ClientOptions), nil
+
+	if cp.Retry.StatusCodes == nil {
+		// remove 429 from the default list of status codes to retry on
+		defaultCodes := exported.DefaultStatusCodesForRetry()
+		subsetCodes := []int{}
```

Comment:
Since DefaultStatusCodesForRetry() returns a new array each time you call it, you could find the 429 and change it to a 0 instead of building a whole new array and then GC'd the returned array.

Reply:
fixed